mlflow/utils/checkpoint_utils.py

```diff
@@ -0,0 +1,174 @@
+import os
+import shutil
+import logging
+import mlflow
+from mlflow.exceptions import MlflowException
+from mlflow.utils.file_utils import create_tmp_dir
+
+from mlflow.utils.mlflow_tags import LATEST_CHECKPOINT_ARTIFACT_TAG_KEY
+from mlflow.utils.autologging_utils import (
+    ExceptionSafeAbstractClass,
+)
+
+
+_logger = logging.getLogger(__name__)
+
+
+class _MlflowModelCheckpointCallbackBase(metaclass=ExceptionSafeAbstractClass):
```

Comment:
we are using this in `pytorch` and `tensorflow` modules, so this cannot be a private class. In MLflow we are abusing the private class/method system, which is hard to clean up, let's not introduce new code that violates the python style guide.